### PR TITLE
Order Return

### PR DESCRIPTION
**Pull request by arijit1111**
Order Returns

### CODE REVIEW

File: Modules/Order/Routes/api.php [hunk 1]
```diff
@@ -21,14 +21,28 @@
...
-        Route::post('/cancel', 'OrderController@cancel')->name('cancel');
+        Route::post('/update', 'OrderController@update')->name('brand.update');
+        Route::post('/split', 'OrderController@split')->name('brand.split');
+        Route::post('/cancel', 'OrderController@cancel')->name('brand.cancel');
+        Route::post('/cancel-request', 'OrderController@cancelRequest')->name('retailer.cancel_request');
         Route::get('/csv', 'OrderController@csv')->name('csv');
+        Route::post('/review', 'OrderController@review')->name('retailer.review');
+        Route::delete('/review/delete/{id}','OrderController@reviewDelete')->name('retailer.review_delete');
+        Route::get('/review/edit/{id}','OrderController@reviewEdit')->name('retailer.review_edit');
+        Route::patch('/review/update/{id}','OrderController@reviewUpdate')->name('retailer.review_update');
```
Comment (smaqeelkazmi) [inline Modules/Order/Routes/api.php]: @arijit1111 Why we are letting customer to update the review which is already posted?

File: Modules/Order/Http/Services/OrderService.php [hunk 2]
```diff
@@ -761,4 +876,168 @@ public function updateBilling(array $request): array
...
+            $policiesStr = implode(',', $returnData["policy_values"]);
+        }
+        $returnData['policies'] = $policiesStr;
+        $returnData['shipping_date'] = date('Y-m-d', strtotime($returnData['shipping_time']));
+        //create order return
+        $orderReturn = new OrderReturn();
+        $orderReturn->fill($returnData);
+        $orderReturn->save();
+
+        if (!empty($returnData['products'])) {
```
Comment (smaqeelkazmi) [inline Modules/Order/Http/Services/OrderService.php]: @arijit1111 User can pass any product ID, and we are not validating either the Product user have passed is a valid and is the User product or not, we need validation.

File: Modules/Order/Http/Services/OrderService.php [hunk 3]
```diff
@@ -761,4 +876,168 @@ public function updateBilling(array $request): array
...
+        $userId = auth()->user()->id;
+        $order = Order::where('order_number', $returnData['order_number'])->first();
+
+        $returnData['order_id'] = $order->id;
+        $policiesStr = '';
+        if (!empty($returnData['policy_values'])) {
+            $policiesStr = implode(',', $returnData["policy_values"]);
+        }
+        $returnData['policies'] = $policiesStr;
+        $returnData['shipping_date'] = date('Y-m-d', strtotime($returnData['shipping_time']));
```
Comment (smaqeelkazmi) [inline Modules/Order/Http/Services/OrderService.php]: @arijit1111 Why we are sending shipping date for order return?

File: Modules/Order/Http/Controllers/OrderController.php [hunk 4]
```diff
@@ -290,4 +363,197 @@ public function csv(Request $request): JsonResponse
...
+
+        if ($user->cannot('cancelRequest', $order)) {
+            return response()->json([
+                'res' => false,
+                'msg' => 'User is not authorized !',
+                'data' => ""
+            ]);
+        }
+
+        $response = $this->orderService->declineOrder($request);
```
Comment (smaqeelkazmi) [inline Modules/Order/Http/Controllers/OrderController.php]: @arijit1111 cannot find the method `declineOrder`

Comment (smaqeelkazmi) [reply Modules/Order/Http/Controllers/OrderController.php]: Please check in the OrderService file here: https://github.com/properweb/Bazar_Repo/blob/2fad6196663258d46b88d7762eecaa424d5038fd/Modules/Order/Http/Services/OrderService.php

**Review by smaqeelkazmi — CHANGES_REQUESTED** [review]
Please see all comments

File: Modules/Order/Routes/api.php (hunk 1 above)
Comment (arijit1111) [reply Modules/Order/Routes/api.php]: edit, update, delete route is added previously, but not working from user perspective, removed it

File: Modules/Order/Http/Services/OrderService.php (hunk 2 above)
Comment (arijit1111) [reply Modules/Order/Http/Services/OrderService.php]: added validation in controller

File: Modules/Order/Http/Services/OrderService.php (hunk 3 above)
Comment (arijit1111) [reply Modules/Order/Http/Services/OrderService.php]: it is returned pickup date

File: Modules/Order/Http/Controllers/OrderController.php (hunk 4 above)
Comment (arijit1111) [reply Modules/Order/Http/Controllers/OrderController.php]: removed form route

**Comment by properweb** [comment]
Please have a look on the newest push, above comments are fixed

**Pull request merged**